src/main/java/com/avispl/symphony/dal/communicator/barco/clickshare/BarcoClickShareCommunicator.java

```diff
@@ -190,23 +294,353 @@ protected void authenticate() throws Exception {
         return Collections.singletonList(extendedStatistics);
     }
 
+    /**
+     * Update supported API version
+     * Used outside of init() method since it's using https endpoint in order to fetch the supported version
+     * which is not possible to do before init() has completed
+     * Might be fetched by both statistics collection and control event ->
+     * if the device adapter has been removed/reloaded - controls on UI may stay stale which will make it
+     * possible to send a control event. If done before the initial statistics call - may lead to an error
+     * (fetching data before the baseUrl which is not possible to set at this stage because of the init() process as well)
+     * @throws Exception during http communication
+     */
+    private void updateSupportedApiVersion() throws Exception {
+        if(StringUtils.isNullOrEmpty(supportedApiVersion)){
+            supportedApiVersion = String.format("%s/", getSupportedVersion());
+        }
+    }
+
+    /**
+     * Performs actions needed for collecting data for devices supporting API V1 (CSE series)
+     * @param statistics map to store statistics data to
+     * @param controls list to add controllable properties to
+     * @throws Exception during http communication
+     */
+    private void v1ApiRoute(Map<String, String> statistics, List<AdvancedControllableProperty> controls) throws Exception {
+        if(!supportedApiVersion.contains(".")){
+            throw new UnsupportedOperationException("The v1 API version without minor version is not supported");
+        }
+        int minorApiVersion = Integer.parseInt(supportedApiVersion.replaceAll("\\/", "").split("\\.")[1]);
+
+        if(minorApiVersion >= 0){

```

Comment:
could it be replaces with switch?

Reply:
well, the thing is that this has got to be cumulative in a way, so that would mean that for version 1.15 all the underlying methods have to be called also. I mean, we could do the switch without break statements there, with a top-down direction, like:
switch(minorApiVersion){
case 14:
v1_14_RequestDeviceInfo(statistics, controls);
case 13:
v1_13_RequestDeviceInfo(statistics, controls);
case 11:
v1_11_RequestDeviceInfo(statistics, controls);
case 8:
v1_8_RequestDeviceInfo(statistics, controls);
....
} 

Is this what you had in mind?